### PR TITLE
[Chat window/CLI] Support test function eager flow without yaml

### PR DESCRIPTION
**Pull request by YingChen1996**
# Description
pf flow test --flow flow:entry_function --ui

CLI:
run cli in Flow dir:   Pf flow test --flow entry: entry_fucntion  -> generate flow flex yaml -> yaml test

Chat window: 
run cli in Flow  dir:   Pf flow test --flow entry: entry_fucntion --ui  : generate flow flex yaml (no contextmanager) -> pass flow flex yaml to ux.  ux call pfs with flow path in the header.

We will not delete genrated flow flex yaml for now.
image

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-devkit/promptflow/_sdk/_utils.py [hunk 1]
```diff
@@ -968,6 +968,19 @@ def overwrite_null_std_logger():
         sys.stderr = sys.stdout
 
 
+def generate_yaml_entry_core(entry: Union[str, PathLike, Callable], code: Path = None):
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/_utils.py:971]: why this ends with core?

File: src/promptflow-devkit/promptflow/_sdk/_utils.py [hunk 2]
```diff
@@ -968,6 +968,19 @@ def overwrite_null_std_logger():
...
+    from promptflow._proxy import ProxyFactory
+
+    executor_proxy = ProxyFactory().get_executor_proxy_cls(FlowLanguage.Python)
+    if callable(entry) or executor_proxy.is_flex_flow_entry(entry=entry):
+        flow_yaml_path, _ = create_temp_flex_flow_yaml_core(entry, code)
+        return flow_yaml_path
+    else:
+        if code:
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/_utils.py:979]: what's this path? add some comment?
when code has value?

File: src/promptflow-devkit/promptflow/_sdk/_utils.py [hunk 3]
```diff
@@ -1000,13 +1010,20 @@ def create_temp_flex_flow_yaml(entry: Union[str, PathLike, Callable], code: Path
     flow_yaml_path = code / FLOW_FLEX_YAML
     existing_content = None
 
+    if flow_yaml_path.exists():
+        logger.warning(f"Found existing {flow_yaml_path.as_posix()}, will not respect it in runtime.")
+        with open(flow_yaml_path, "r", encoding=DEFAULT_ENCODING) as f:
+            existing_content = f.read()
+    with open(flow_yaml_path, "w", encoding=DEFAULT_ENCODING) as f:
+        dump_yaml({"entry": entry}, f)
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/_utils.py:1018]: will we merge the exsiting flow yaml? or write a yaml with one line on entry?

File: src/promptflow-devkit/promptflow/_cli/_pf/_flow.py [hunk 4]
```diff
@@ -510,18 +511,21 @@ def _test_flow_multi_modal(args, pf_client):
         from promptflow._sdk._tracing import _invoke_pf_svc
 
         # Todo: use base64 encode for now, will consider whether need use encryption or use db to store flow path info
-        def generate_url(flow_path, port):
+        def generate_url(flow_path, port, enable_internal_features=False):
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_cli/_pf/_flow.py:514]: will this PR also address case that without --ui?

Comment (YingChen1996) [reply src/promptflow-devkit/promptflow/_cli/_pf/_flow.py:514]: yeap

File: src/promptflow-devkit/promptflow/_sdk/_utils.py (hunk 3 above)
Comment (YingChen1996) [reply src/promptflow-devkit/promptflow/_sdk/_utils.py:1018]: For flow test case, write a yaml with one line on entry

File: src/promptflow-devkit/promptflow/_sdk/_utils.py (hunk 1 above)
Comment (YingChen1996) [reply src/promptflow-devkit/promptflow/_sdk/_utils.py]: changed name

**Review by wangchao1230 — APPROVED** [review]

**Review by elliotzh — APPROVED** [review]

**Pull request merged**